backend/src/main/java/fk/prof/backend/model/request/RecordedProfileHeaderParser.java

```diff
@@ -0,0 +1,106 @@
+    this.maxAllowedBytesForRecordingHeader = maxAllowedBytesForRecordingHeader;
+  }
+
+  /**
+   * Returns true if all fields of header have been read and checksum validated, false otherwise
+   *
+   * @return returns if header has been parsed or not
+   */
+  public boolean isParsed() {
+    return this.headerParsed;
+  }
+
+  /**
+   * Returns {@link RecordedProfileHeader} if {@link #isParsed()} is true, null otherwise
+   *
+   * @return
+   */
+  public RecordedProfileHeader get() {
+    if (!this.headerParsed) {
+      return null;
+    }
+    return new RecordedProfileHeader(this.encodedVersion, this.recordingHeader);
+  }
+
+  /**
+   * Reads buffer and updates internal state with parsed fields. Returns the starting unread position in outputstream
+   *
+   * @param codedInputStream
+   * @return starting unread position in buffer
+   */
+  public int parse(CodedInputStream codedInputStream, Buffer underlyingBuffer, int currentPos) throws HttpFailure {
+    try {
+      if (!headerParsed) {
+        if (encodedVersion == null) {
+          encodedVersion = codedInputStream.readUInt32();
+          currentPos = updateChecksumAndGetNewPos(underlyingBuffer, codedInputStream, currentPos);
+        }
+
+        if (headerLength == null) {
+          headerLength = codedInputStream.readUInt32();
+          if (headerLength < 1 || headerLength > maxAllowedBytesForRecordingHeader) {
+            throw new HttpFailure("Allowed range for recording header length is 1B to " + maxAllowedBytesForRecordingHeader + "B", 400);
+          }
+          currentPos = updateChecksumAndGetNewPos(underlyingBuffer, codedInputStream, currentPos);
+        }
+
+        if (recordingHeader == null) {
+          if (headerLength > (underlyingBuffer.length() - currentPos)) {
+            return currentPos;
+          }
+
+          int oldLimit = codedInputStream.pushLimit(headerLength);
+          try {
+            recordingHeader = Recorder.RecordingHeader.parseFrom(codedInputStream);
+          } catch (IOException ex) {
+            //Running buffer has sufficient bytes present for reading recording header. If exception is thrown while parsing, send error response
+            throw new HttpFailure("Error while parsing recording header", 400);
+          }
+          codedInputStream.popLimit(oldLimit);
+          currentPos = updateChecksumAndGetNewPos(underlyingBuffer, codedInputStream, currentPos);
+        }
+
+        if (checksumValue == null) {
+          checksumValue = codedInputStream.readUInt32();
+          currentPos = codedInputStream.getTotalBytesRead();
+          if ((int) headerChecksum.getValue() != checksumValue) {
+            throw new HttpFailure("Checksum of header does not match", 400);
+          }
+          headerParsed = true;
+        }
+      }
+    } catch (IOException ex) {
+      //NOTE: Ignore this exception. Can come because incomplete request has been received. Chunks can be received later
```

Comment:
This is thrown by protobuf object's parsefrom method. I will check if it throws specific IOException in case of missing bytes